source/validate.cpp

```diff
@@ -518,6 +436,109 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+// Allows for piping of multiple functions which return spv_result_t
+spv_result_t operator||(spv_result_t lhs, spv_result_t rhs) {
```

Comment:
Not sure why this is required, given that `spv_result_t` implicitly converts to `bool`.  What happens if you delete this function?